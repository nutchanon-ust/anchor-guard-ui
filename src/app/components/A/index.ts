import styled from 'styled-components/macro';

export const A = styled.a`
  color: ${p => p.theme.primary};
  text-decoration: none;

  &:hover {
    text-decoration: underline;
    opacity: 0.8;
  }

  &:active {
    opacity: 0.4;
  }
`;
