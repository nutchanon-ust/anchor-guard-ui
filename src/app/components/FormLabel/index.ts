import styled from 'styled-components/macro';

export const FormLabel = styled.label`
  text-transform: uppercase;
  font-weight: normal;
  margin: 0;
  padding: 0;
  color: ${p => p.theme.textSecondary};
  font-size: 0.75rem;
`;
