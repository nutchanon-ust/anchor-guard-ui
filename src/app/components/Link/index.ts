import styled from 'styled-components/macro';
import { Link as RouterLink } from 'react-router-dom';

export const Link = styled(RouterLink)`
  color: ${p => p.theme.primary};
  text-decoration: none;

  &:hover {
    text-decoration: underline;
    opacity: 0.8;
  }

  &:active {
    opacity: 0.4;
  }
`;
