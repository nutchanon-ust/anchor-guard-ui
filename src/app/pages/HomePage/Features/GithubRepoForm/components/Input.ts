import styled from 'styled-components/macro';

export const Input = styled.input`
  border: 2px solid ${p => p.theme.border};
  border-radius: 4px;
  color: ${p => p.theme.text};
  display: block;
  width: 100%;
  font-size: 1rem;
  font-weight: 500;
  line-height: 1.5;
  color: ${p => p.theme.text};
  background-color: ${p => p.theme.background};
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  outline: none;
  height: 2.75rem;
  padding: 0 0.75rem;

  &::placeholder {
    font-weight: 300;
    color: ${p => p.theme.textSecondary};
  }

  &:focus {
    border-color: ${p => p.theme.primary};
    box-shadow: 0 0 0 3px
      ${p =>
        p.theme.primary.replace(
          /rgba?(\(\s*\d+\s*,\s*\d+\s*,\s*\d+)(?:\s*,.+?)?\)/,
          'rgba$1,0.2)',
        )};
  }
`;
