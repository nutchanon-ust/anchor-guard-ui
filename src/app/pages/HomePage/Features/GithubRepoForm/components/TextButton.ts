import styled from 'styled-components/macro';

export const TextButton = styled.button`
  background: none;
  outline: none;
  padding: 0;
  margin: 0;
  border: none;
  color: ${p => p.theme.primary};
  cursor: pointer;

  &:hover {
    opacity: 0.8;
    text-decoration: underline;
  }

  &:active {
    opacity: 0.4;
  }
`;
