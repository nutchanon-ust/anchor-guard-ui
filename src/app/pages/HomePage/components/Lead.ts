import styled from 'styled-components/macro';

export const Lead = styled.p`
  font-size: 1.5rem;
  font-weight: 300;
  line-height: 1.5;
  color: ${p => p.theme.textSecondary};
  margin: 0 0 1.5rem 0;

  strong {
    color: ${p => p.theme.text};
  }
`;
