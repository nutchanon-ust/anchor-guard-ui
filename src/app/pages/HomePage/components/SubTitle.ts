import styled from 'styled-components/macro';

export const SubTitle = styled.h3`
  font-size: 1.25rem;
  margin: 0;
  color: ${p => p.theme.text};
`;
