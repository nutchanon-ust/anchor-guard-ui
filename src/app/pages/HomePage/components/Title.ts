import styled from 'styled-components/macro';

export const Title = styled.h1`
  font-size: 32px;
  font-weight: bold;
  color: ${p => p.theme.text};
  margin: 1rem 0;
`;
