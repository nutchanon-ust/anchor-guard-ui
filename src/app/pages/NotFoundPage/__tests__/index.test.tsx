import * as React from 'react';
import { NotFoundPage } from '..';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { themes } from 'styles/theme/themes';
import { HelmetProvider } from 'react-helmet-async';
import renderer from 'react-test-renderer';
import { Link } from 'app/components/Link';

const renderPage = () =>
  renderer.create(
    <MemoryRouter>
      <ThemeProvider theme={themes.light}>
        <HelmetProvider>
          <NotFoundPage />
        </HelmetProvider>
      </ThemeProvider>
    </MemoryRouter>,
  );

describe('<NotFoundPage />', () => {
  it('should match snapshot', () => {
    const notFoundPage = renderPage();
    expect(notFoundPage.toJSON()).toMatchSnapshot();
  });

  it('should should contain Link', () => {
    const notFoundPage = renderPage();
    expect(notFoundPage.root.findByType(Link)).toBeDefined();
  });
});
