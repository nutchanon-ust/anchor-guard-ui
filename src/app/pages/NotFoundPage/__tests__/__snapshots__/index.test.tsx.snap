// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFoundPage /> should match snapshot 1`] = `
Array [
  .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  font-size: 1.25rem;
  color: rgba(58,52,51,1);
  font-weight: bold;
  margin-right: 1rem;
}

.c5 {
  font-size: 0.875rem;
  color: rgba(58,52,51,0.7);
  font-weight: normal;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -1rem;
}

.c7 {
  color: rgba(215,113,88,1);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.25rem 1rem;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7:hover {
  opacity: 0.8;
}

.c7:active {
  opacity: 0.4;
}

.c7 .icon {
  margin-right: 0.25rem;
}

.c2 {
  width: 960px;
  margin: 0 auto;
  padding: 0 1.5rem;
  box-sizing: content-box;
}

.c0 {
  box-shadow: 0 1px 0 0 rgba(58,52,51,0.05);
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  width: 100%;
  background-color: rgba(255,255,255,1);
  z-index: 2;
}

.c0 .c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@supports (-webkit-backdrop-filter:blur(10px)) or (backdrop-filter:blur(10px)) {
  .c0 {
    -webkit-backdrop-filter: blur(10px);
    backdrop-filter: blur(10px);
    background-color: rgba(255,255,255,0.75);
  }
}

<header
    className="c0"
  >
    <div
      className="c1 c2"
    >
      <div
        className="c3"
      >
        <div
          className="c4"
        >
          React Boilerplate
        </div>
        <div
          className="c5"
        >
          Create React App Template
        </div>
      </div>
      <nav
        className="c6"
      >
        <a
          className="c7"
          href="https://cansahin.gitbook.io/react-boilerplate-cra-template/"
          rel="noopener noreferrer"
          target="_blank"
          title="Documentation Page"
        >
          <svg>
            documentation-icon.svg
          </svg>
          Documentation
        </a>
        <a
          className="c7"
          href="https://github.com/react-boilerplate/react-boilerplate-cra-template"
          rel="noopener noreferrer"
          target="_blank"
          title="Github Page"
        >
          <svg>
            github-icon.svg
          </svg>
          Github
        </a>
      </nav>
    </div>
  </header>,
  .c2 {
  font-size: 1rem;
  line-height: 1.5;
  color: rgba(58,52,51,0.7);
  margin: 0.625rem 0 1.5rem 0;
}

.c3 {
  color: rgba(215,113,88,1);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  opacity: 0.8;
}

.c3:active {
  opacity: 0.4;
}

.c0 {
  height: calc(100vh - 4rem);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 320px;
}

.c1 {
  margin-top: -8vh;
  font-weight: bold;
  color: rgba(58,52,51,1);
  font-size: 3.375rem;
}

.c1 span {
  font-size: 3.125rem;
}

@supports (-webkit-backdrop-filter:blur(10px)) or (backdrop-filter:blur(10px)) {

}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      4
      <span
        aria-label="Crying Face"
        role="img"
      >
        ðŸ˜¢
      </span>
      4
    </div>
    <p
      className="c2"
    >
      Page not found.
    </p>
    <a
      className="c3"
      href="/"
      onClick={[Function]}
    >
      Return to Home Page
    </a>
  </div>,
]
`;
